python/spear/sp_engine.py

```diff
@@ -0,0 +1,271 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import msgpackrpc
+import os
+import psutil
+import spear
+from subprocess import Popen
+import sys
+import time
+
+class SpEngine():
```

Comment:
The responsibilities of this Python class are slightly broader than the `SpEngine` C++ class. For example, the Python class needs to know about launching the underlying `SpearSim` executable, PID values, command-line args, etc, whereas the `SpEngine` C++ class doesn't need to know about any of that. So the two classes are not perfect mirror images of each other. Therefore, I prefer the name `Instance` for this Python class.